packit_service/worker/testing_farm.py

```diff
@@ -126,27 +126,68 @@ def _artifact(
         return artifact
 
     def _payload(
-        self, chroot: str, artifact: Optional[Dict[str, Union[List[str], str]]] = None
+        self,
+        chroot: str,
+        artifact: Optional[Dict[str, Union[List[str], str]]] = None,
+        build_id: Optional[int] = None,
     ) -> dict:
-        """
+        """Prepare a Testing Farm request payload.
+
         Testing Farm API: https://testing-farm.gitlab.io/api/
 
         Currently we use the same secret to authenticate both,
         packit service (when sending request to testing farm)
         and testing farm (when sending notification to packit service's webhook).
         We might later use a different secret for those use cases.
 
+        Args:
+            chroot: Target TF chroot.
+            build_id: ID of the related copr build.
+            artifact: Optional artifacts, e.g. list of package NEVRAs
         """
         distro, arch = self.chroot2distro_arch(chroot)
         compose = self.distro2compose(distro, arch)
         fmf = {"url": self.fmf_url}
         if self.fmf_ref:
             fmf["ref"] = self.fmf_ref
 
+        if build_id is not None:
+            copr_build = CoprBuildModel.get_by_build_id(build_id)
+            build_log_url = copr_build.build_logs_url
+            srpm_build = copr_build.get_srpm_build()
+            srpm_url = srpm_build.url
+            nvr_data = copr_build.built_packages[0]
+            nvr = f"{nvr_data['name']}-{nvr_data['version']}-{nvr_data['release']}"
+        else:
+            build_log_url = nvr = srpm_url = None
+
+        predefined_environment = {
+            "PACKIT_FULL_REPO_NAME": self.project.full_repo_name,
+            "PACKIT_UPSTREAM_NAME": self.job_config.upstream_package_name,
+            "PACKIT_UPSTREAM_URL": self.job_config.upstream_project_url,
+            "PACKIT_DOWNSTREAM_NAME": self.job_config.downstream_package_name,
+            "PACKIT_DOWNSTREAM_URL": self.job_config.downstream_project_url
+            if self.job_config.downstream_package_name
+            else None,
+            "PACKIT_PACKAGE_NAME": self.job_config.downstream_package_name,
```

Comment:
As I see it now, we probably don't need both `PACKIT_DOWNSTREAM_NAME` and `PACKIT_PACKAGE_NAME`.

Reply:
Which one do you find better? I think that `PACKIT_DOWNSTREAM_NAME` makes a bit more sense since we also have `PACKIT_UPSTREAM_NAME`. On the other hand, I don't see anything wrong with keeping both and letting the users decide whichever they prefer (some may prefer the term `downstream`, others `package`).

Reply:
I agree, I would use `PACKIT_DOWNSTREAM_NAME` or both. I'll leave it up to you...;)

Reply:
Lets keep both then. I am merging this PR and will hopefully be able to see on staging if everything is fine.